nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -985,3 +986,34 @@ def _fetch_next_batch(self, iterator: Iterator) -> None:
             assert isinstance(dataloader, Sized)  # `_has_len` is True
             self.done = self.fetched >= len(dataloader)
         self.on_fetch_end(batch, start_output)
+
+
+class CustomProgressBar(TQDMProgressBar):
+    """
+    Add a CustomProgressBar to remove 's/it' and add train_step_timing for megatron models
+    """
+
+    def __init__(self, cfg):
+        super().__init__()
+        self.cfg = cfg
+
+    def init_train_tqdm(self):
+        """
+        Override progress bar total to reflect total microbatches as PTL 2.0 progress bar indicates total microbatches, and bar format 
+        to remove 's/it' from the progress bar for megatron models
+        """
+        self.bar = super().init_train_tqdm()
+        # TODO athitten: Remove the below line once PTL fixes the logic for progress bar with dataloader_iter
+        self.bar.total = self.cfg.trainer.max_steps * get_num_microbatches()
```

Comment:
why not have the total as only `self.cfg.trainer.max_steps` and then ensure that we count 1 num_microbatches worth of data extraction from the dataloader as 1 step?

this would make things clearer for an end user, currently when total = max_steps * get_num_microbatches, the total in the progressbar would not make any sense to the user

Reply:
@aklife97 this is something that is broken in progress bar with lightning 2.0 and ideally it should be the way you described. This was reported to PTL and they said they'll fix it. I am unsure at this point if we can have the behavior you mentioned through the CustomProgressBar class, I will check on this.
Let me give you some context behind why I decided to have it this way: Currently PTL uses max_steps in its progress bar but displays all micro batch iters. So after the iterations reach max_steps, it displays only the iterations separately without progress bar. SS below:
![image](https://github.com/NVIDIA/NeMo/assets/47577437/7383a676-540a-4a22-8f36-61a10a255503)
So when I overwrote the bar format to remove `'s/it'` the progress bar started displaying` iter_num/?` after max_steps were reached. To remove the `?` in the display as it would look ugly I decided to have bar total as the total microbtaches since PTL's progress bar is currently indicating all microbatch iterations.

Reply:
@aklife97 this is fixed. We now override PTL 2.0's default behavior of updating per microbatch and instead update per global batch.